uuid.go

```diff
@@ -112,13 +117,13 @@ func ParseUUID(s string) (uuid UUID, err error) {
 // String returns a human-readable version of this UUID, such as
 // 00001234-0000-1000-8000-00805f9b34fb.
 func (uuid UUID) String() string {
-	// TODO: make this more efficient.
-	s := ""
+	var s strings.Builder
+	s.Grow(36)
```

Comment:
This can probably be sped up even more, at the cost of readability. You can make a byte array (`make([]byte, 36)`), fill it, and then convert it to a string. That should result in only a single heap allocation (to allocate the string) if the compiler is able to optimize away the `make` to a stack allocation.

I'm not sure that's worth it, but it could be an interesting optimization.